api-2.3/src/main/java/org/openmrs/module/htmlformentry/HtmlFormEntryUtil2_3.java

```
@@ -0,0 +1,39 @@
+package org.openmrs.module.htmlformentry;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang.math.NumberUtils;
+import org.openmrs.FormRecordable;
+
+/**
+ * HTML Form Entry utility methods with Core 2.3 compatibility
+ */
+public class HtmlFormEntryUtil2_3 {
+	
+	/**
+	 * Returns the control id part out of an OpenMRS data object's form namespace and path.
+	 * Eg: "my_condition_tag" out of "HtmlFormEntry^MyForm.1.0/my_condition_tag-0"
+	 *
+	 * @param openmrsData The form recordable OpenMRS data object
+	 * @return The control id
+	 */
+	public static String getControlId(FormRecordable openmrsData) {
+		
+		// Validate if getFormFieldPath exists.
+		if (StringUtils.isEmpty(openmrsData.getFormFieldPath())) {
+			throw new IllegalStateException(
+			        "A form recordable object was found to have no form namespace and path set, its control id in the form could not be determined.");
+		}
```

for what it's worth, when we expand Obs to support a form path, we will need change this, because we need to support legacy forms/obs that don't have a control Id.

---

@icrc-loliveira I suggest that we return `null` early then, and let the caller throw exceptions. In our case the `IllegalStateException` will this be thrown in `ConditionElement`.

Please adjust the JavaDoc accordingly and add a test case in `ConditionElementTest`.